contracts/GoodGhostingPolygonCurve.sol

```diff
@@ -0,0 +1,674 @@
+        require(player.amountPaid > 0, "Player does not exist");
+        require(!player.withdrawn, "Player has already withdrawn");
+        player.withdrawn = true;
+        activePlayersCount = activePlayersCount.sub(1);
+        if (winnerCount > 0 && player.isWinner) {
+            winnerCount = winnerCount.sub(uint256(1));
+            player.isWinner = false;
+            winners[player.winnerIndex] = address(0);
+        }
+
+        // In an early withdraw, users get their principal minus the earlyWithdrawalFee % defined in the constructor.
+        uint256 withdrawAmount = player.amountPaid.sub(
+            player.amountPaid.mul(earlyWithdrawalFee).div(100)
+        );
+        // Decreases the totalGamePrincipal on earlyWithdraw
+        totalGamePrincipal = totalGamePrincipal.sub(player.amountPaid);
+        uint256 currentSegment = getCurrentSegment();
+
+        // Users that early withdraw during the first segment, are allowed to rejoin.
+        if (currentSegment == 0) {
+            player.canRejoin = true;
+        }
+
+        emit EarlyWithdrawal(msg.sender, withdrawAmount, totalGamePrincipal);
+
+        if (poolType == 0) {
+            uint256[numAaveTokens] memory amounts;
+            for (uint256 i = 0; i < numAaveTokens; i++) {
+                if (i == inboundTokenIndexUint) {
+                    amounts[i] = withdrawAmount;
+                } else {
+                    amounts[i] = 0;
+                }
+            }
+            uint256 poolWithdrawAmount = pool.calc_token_amount(amounts, true);
+
+            if (gauge.balanceOf(address(this)) < poolWithdrawAmount) {
+                poolWithdrawAmount = gauge.balanceOf(address(this));
+            }
+
+            gauge.withdraw(poolWithdrawAmount, false);
+
+            pool.remove_liquidity_one_coin(
+                poolWithdrawAmount,
+                inboundTokenIndexInt,
+                _minAmount,
+                true
+            );
+        } else if (poolType == 1) {
+            uint256[numAtricryptoTokens] memory amounts;
+            for (uint256 i = 0; i < numAtricryptoTokens; i++) {
+                if (i == inboundTokenIndexUint) {
+                    amounts[i] = withdrawAmount;
+                } else {
+                    amounts[i] = 0;
+                }
+            }
+            uint256 poolWithdrawAmount = pool.calc_token_amount(amounts, true);
+
+            if (gauge.balanceOf(address(this)) < poolWithdrawAmount) {
+                poolWithdrawAmount = gauge.balanceOf(address(this));
+            }
+
+            gauge.withdraw(poolWithdrawAmount, false);
+
+            require(
+                lpToken.approve(address(pool), poolWithdrawAmount),
+                "Fail to approve allowance to pool"
+            );
+            pool.remove_liquidity_one_coin(
+                poolWithdrawAmount,
+                inboundTokenIndexUint,
+                _minAmount
+            );
+        }
```

Comment:
this can probably be simplified. you can use the if condition to set `numAaveTokens` or `numAtricryptoTokens` to a variable and run the loop using that.

Then use the if condition to call `remove_liquidity_one_coing accordingly.

Everything else seems the same.

Reply:
answered above